File: app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/AnalyticsService.java
package openfoodfacts.github.scrachx.openfood.utils;

import io.sentry.core.Sentry;
import openfoodfacts.github.scrachx.openfood.BuildConfig;

public class AnalyticsService {
    private AnalyticsService() {
    }

    public static void init() {
        Sentry.configureScope(scope -> scope.setTag("flavor", BuildConfig.FLAVOR));
    }

    public static void log(String key, String value) {
        Sentry.setTag(key, value);
    }

    public static void record(Exception exception) {
        Sentry.captureException(exception);
    }
}

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/AnalyticsService.kt
package openfoodfacts.github.scrachx.openfood.utils

import io.sentry.Scope
import io.sentry.Sentry
import openfoodfacts.github.scrachx.openfood.BuildConfig

object AnalyticsService {
    @JvmStatic
    fun init() {
        Sentry.configureScope { scope: Scope -> scope.setTag("flavor", BuildConfig.FLAVOR) }
    }

    fun log(key: String, value: String) {
        Sentry.setTag(key, value)
    }

    fun record(exception: Exception?) {
        Sentry.captureException(exception!!)
    }
}